contracts/sources/obligation.move

```diff
@@ -349,26 +427,57 @@ module suilend::obligation {
         );
         obligation.weighted_borrowed_value_usd = add(
             obligation.weighted_borrowed_value_usd, 
-            mul(borrow_market_value, borrow_weight(config(reserve)))
+            mul(borrow_market_value, borrow_weight)
         );
         obligation.weighted_borrowed_value_upper_bound_usd = add(
             obligation.weighted_borrowed_value_upper_bound_usd, 
-            mul(borrow_market_value_upper_bound, borrow_weight(config(reserve)))
+            mul(borrow_market_value_upper_bound, borrow_weight)
         );
 
         let user_reward_manager = vector::borrow_mut(&mut obligation.user_reward_managers, borrow.user_reward_manager_index);
         liquidity_mining::change_user_reward_manager_share(
-            reserve::borrows_pool_reward_manager_mut(reserve),
+            reserve::borrows_pool_reward_manager_mut(borrow_reserve),
             user_reward_manager,
             liability_shares(borrow),
             clock
         );
 
         assert!(is_healthy(obligation), EObligationIsNotHealthy);
 
-        if (isolated(config(reserve)) || obligation.borrowing_isolated_asset) {
+        if (isolated(config(borrow_reserve)) || obligation.borrowing_isolated_asset || is_emode) {
             assert!(vector::length(&obligation.borrows) == 1, EIsolatedAssetViolation);
         };
+
+        if (is_emode) {
```

Comment:
can you just call obligation::refresh() here instead?

Reply:
done